lib/Runtime/Library/JavascriptProxy.cpp

```diff
@@ -919,11 +919,14 @@ namespace Js
         Var trapResult = getEnumeratorMethod->CallFunction(arguments);
         threadContext->SetImplicitCallFlags((Js::ImplicitCallFlags)(saveImplicitCallFlags | ImplicitCall_Accessor));
 
-        if (!JavascriptOperators::IsObject(trapResult))
+        if (!JavascriptOperators::IsArray(trapResult))
         {
-            JavascriptError::ThrowTypeError(scriptContext, JSERR_InconsistentTrapResult, _u("enumerate"));
+            JavascriptError::ThrowTypeError(scriptContext, JSERR_InconsistentTrapResult, _u("ownKeys"));
         }
-        *enumerator = IteratorObjectEnumerator::Create(scriptContext, trapResult);
+
+        Var arrayIterator = JavascriptOperators::GetIterator(RecyclableObject::FromVar(trapResult), scriptContext);
+        *enumerator = IteratorObjectEnumerator::Create(scriptContext, arrayIterator);
+
         return TRUE;
```

Comment:
instead of iterators, it's easier to use enumerator. You can see the code in JavascriptObject::CreateKeysHelper, in similar pattern.